src/main/java/com/amazon/reviews/service/impl/CsvToReviewDtoParsingServiceImpl.java

```diff
@@ -0,0 +1,62 @@
+package com.amazon.reviews.service.impl;
+
+import com.amazon.reviews.exception.FileProcessingException;
+import com.amazon.reviews.model.dto.ReviewDto;
+import com.amazon.reviews.service.ParsingService;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CsvToReviewDtoParsingServiceImpl implements ParsingService<ReviewDto> {
+    private static final int ID_INDEX = 0;
+    private static final int PRODUCT_ID_INDEX = 1;
+    private static final int USER_ID_INDEX = 2;
+    private static final int PROFILE_NAME_INDEX = 3;
+    private static final int HELPFULNESS_NUMERATOR_INDEX = 4;
+    private static final int HELPFULNESS_DENOMINATOR_INDEX = 5;
+    private static final int SCORE_INDEX = 6;
+    private static final int TIME_INDEX = 7;
+    private static final int SUMMARY_INDEX = 8;
+    private static final int TEXT_INDEX = 9;
+    private static final int LAST_CSV_COMMA = 9;
+    private static final int PARSED_LINE_SIZE = 10;
+
+    @Override
+    public ReviewDto parse(String line) {
+        String reviewDetails;
+        String reviewText;
+        int commaIndex;
+        String[] parsedLine = new String[PARSED_LINE_SIZE];
+        try {
+            commaIndex = StringUtils.ordinalIndexOf(line, ",", LAST_CSV_COMMA);
```

Comment:
I can see that you have `univocity-parsers` library in your pom.xml

this should work as following:
you have your data: `String[] data = csvParser.parseLine(line);` and then can call each field by its index

the only problem is that you might need to call `new CsvParserSettings().setMaxCharsPerColumn()` method and then pass these settings to CsvParser constructor

please check - will this work?

Reply:
I can't make it work. The review text is cut off at the first 'internal' comma 😢